### PR TITLE
adds migration to copy application_id to application_hbx_id

### PR DESCRIPTION
**Pull request by saikumar9**
# PR Checklist

Please check if your PR fulfills the following requirements:
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bug fixes/features), and they use `let` helpers and `before` blocks.
- [ ] For all UI changes, there is Cucumber coverage.
- [ ] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, the reasoning is documented in the PR and code.
- [ ] Any endpoint modified in the PR only responds to the expected MIME types.
- [x] For all scripts or rake tasks, how to run them is documented in both the PR and the code.
- [ ] There are no inline styles added.
- [ ] There is no inline JavaScript added.
- [ ] There is no hard-coded text added/updated in helpers/views/JavaScript. New/updated translation strings do not include markup/styles unless there is supporting documentation.
- [ ] Code does not use `.html_safe`.
- [ ] All images added/updated have alt text.
- [x] Does not bypass RuboCop rules in any way.

# PR Type
What kind of change does this PR introduce?:

- [ ] Bugfix
- [ ] Feature (requires Feature flag)
- [x] Data fix, Migration, or Report (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: [IVL Product Development - 188509314](https://www.pivotaltracker.com/story/show/188509314)

# A brief description of the changes:

Current behavior: N/A

New behavior: Adds data migration to copy application_id to application_hbx_id.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and indicate which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context

```ruby
# Command to trigger the script:
#   CLIENT=me bundle exec rails runner script/tax_household_groups/copy_application_id_to_application_hbx_id.rb
```

### CODE REVIEW

File: script/tax_household_groups/copy_application_id_to_application_hbx_id.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#   CLIENT=me bundle exec rails runner script/tax_household_groups/copy_application_id_to_application_hbx_id.rb
+
+p '********** STARTING - Script to copy the application id to the application hbx id for eligible tax household groups. **********'
+
+# Calls the operation to copy the application id to the application hbx id for all the tax household groups.
+# If the job is successful, it logs the success message and instructions.
+# If the job fails, it logs the failure message.
+#
+# @return [void]
+result = ::Operations::Migrations::TaxHouseholdGroups::CopyApplicationIdToApplicationHbxId.new.call
```
Comment (mdkaraman) [inline script/tax_household_groups/copy_application_id_to_application_hbx_id.rb:15]: Just a thought-- should we bench mark the time to complete here?  I know it's helpful for Prod deployment to know how long the migration roughly takes.

**Review by mdkaraman — DISMISSED** [review]
Looks good!  Added one thought/suggestion but not a blocker necessarily.

**Review by saipraveen18 — DISMISSED** [review]
lgmt 👍

**Review by saipraveen18 — APPROVED** [review]
lgtm 👍

**Review by mdkaraman — APPROVED** [review]
Looks good!  Very useful addition of the benchmark cache class.  Taking note of this so I can use it in the future.

**Pull request merged**